internal/service/alarms/internal/serviceconfig/serviceconfig.go

```diff
@@ -0,0 +1,300 @@
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/openshift-kni/oran-o2ims/internal/controllers/utils"
+	"github.com/openshift-kni/oran-o2ims/internal/service/alarms/internal/db/models"
+	"github.com/stephenafamo/bob/dialect/psql"
+	"github.com/stephenafamo/bob/dialect/psql/dm"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	cleanupCronJobName   = "alarms-server-events-cleanup"
+	cleanupConfigMapName = "alarms-server-events-cleanup-sql"
+	cleanScriptDir       = "/scripts"
+	cleanScriptName      = "cleanup.sql"
+)
+
+var (
+	cronSchedule    = "0 * * * *" // run at the start of every hour e.g 2:00, 3:00 etc
+	jobBackoffLimit = int32(3)
+	successLimit    = int32(3)
+	failureLimit    = int32(1)
+	commonLabels    = map[string]string{
+		"app": "alarms-server",
+	}
+)
+
+// StartCleanupCronJob starts a cronjob with all the required resources to do alarms events cleanup.
+func StartCleanupCronJob(ctx context.Context, sc *models.ServiceConfiguration, pg *pgxpool.Pool, hc client.Client) error {
+	// Call cleanup first. This will ensure there's no drift between configmap updates and what the cronjob is actually reading (stale jobs).
+	// Alternatively there's config reloader pattern (set a watch on cm) but this "delete + create" is much simpler given minimal performance impact + how rarely it will be used.
+	if err := CleanupResources(ctx, hc); err != nil {
+		return fmt.Errorf("could clneaup exisiting resources: %w", err)
+	}
+
+	// Create CM with sql
+	configMap, err := generateConfigMapWithSql(sc)
+	if err != nil {
+		return fmt.Errorf("generate sql configmap: %w", err)
+	}
+	if err := hc.Create(ctx, &configMap); err != nil {
+		return fmt.Errorf("failed to apply configmap: %w", err)
+	}
+
+	// Create CronJob that reads the CM
+	cronJob, err := generateCronJob(ctx, pg, hc, configMap.Name)
+	if err != nil {
+		return fmt.Errorf("create cronjob spec: %w", err)
+	}
+	if err := hc.Create(ctx, &cronJob); err != nil {
+		return fmt.Errorf("failed to apply cronjob: %w", err)
+	}
+
+	slog.Info("Successfully created cronjob and other required resources for alarms events cleanup", "cronjob", cronJob.Name, "configmap", configMap.Name)
+	return nil
+}
+
+// generateConfigMapWithSql a simple configMap to hold the sql command that is called from cronjob
+func generateConfigMapWithSql(sc *models.ServiceConfiguration) (corev1.ConfigMap, error) {
+	sql, err := getCleanUpPgSQL(sc)
+	if err != nil {
+		return corev1.ConfigMap{}, fmt.Errorf("failed to generate cleanup sql for configmap: %w", err)
+	}
+
+	return corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cleanupConfigMapName,
+			Namespace: utils.DefaultNamespace,
```

Comment:
We should try to get our namespace from our env.

Reply:
using env